custom/icds_reports/management/commands/push_pmo_data.py

```diff
@@ -0,0 +1,141 @@
+            'cbe_conducted',
+            'vhnd_conducted',
+        )
+
+
+        pmo_month_payload = []
+        for data in pmo_data_query:
+            packet = dict()
+            packet.update(self.STATIC_PROJECT_DATA)
+            packet['state_code'] = data['state_site_code']
+            packet['district_code'] = data['district_site_code'][2:]
+            packet['cnt1'] = int(data['num_launched_awcs'])
+            packet['cnt2'] = round(data['bf_at_birth'] / float(data['born_in_month']), 5)
+            packet['cnt3'] = round(data['cf_initiation_in_month'] / float(data['cf_initiation_eligible']), 5)
+            packet['cnt4'] = round(data['wer_weighed'] / float(data['wer_eligible']), 5)
+            packet['cnt5'] = int(data['cbe_conducted'])
+            packet['cnt6'] = int(data['vhnd_conducted'])
+            packet['yr'] = month.year
+            packet['mnth'] = month.month
+            packet['datadt'] = (month + relativedelta(months=1, seconds=-1)).strftime('%m/%d/%Y')  # month end
+            packet_string = json.dumps(packet)
+
+            pmo_month_payload.append(self.encrypt_string(packet_string))
+
+        return pmo_month_payload
+
+    def encrypt_string(self, input_string):
+        input_string += f"|dt={datetime.now()}"
+        input_string += f"|checksum={self.get_string_hash(input_string)}"
+        cipher = AES.new(self.encryption_key, AES.MODE_CBC, iv=self.encryption_key)
+        input_string_bytes = input_string.encode()
+        ct_bytes = cipher.encrypt(pad(input_string_bytes, 16))
+        return b64encode(ct_bytes).decode('utf-8')
+
+    def send_data(self, missing_data_list):
+        data_payload = {
+            'IP': {
+                "mcode": 0, # TODO: Will be updated as NIC gives us this
+                "state_code": 999,
+                "dept_code": self.STATIC_PROJECT_DATA['dept_code'],
+                "project_code": self.STATIC_PROJECT_DATA['project_code'],
+                "sec_code": self.STATIC_PROJECT_DATA['sector_code']
+            },
+            "Records": missing_data_list
+        }
+        headers = {'Content-Type': "application/json"}
+        response = requests.post(self.data_push_api, data=data_payload, headers=headers)
+        try:
+            json_content = json.loads(response.content)
+            print(json_content['Status'])
+            print(json_content['Message'])
+
+        except json.decoder.JSONDecodeError:
+            print(response.status_code)
+            print(response.content)
+
+    def add_arguments(self, parser):
+        parser.add_argument('encryption_key', help="Encryption key with which data has to be encrypted.",)
+
+    def add_arguments(self, parser):
+        parser.add_argument('date_range_api', help="API URL to fetch the pendency",)
+    def add_arguments(self, parser):
+        parser.add_argument('data_push_api', help="API URL to push the data",)
+
+    def handle(self, encryption_key, date_range_api, data_push_api, *args, **options):
```

Comment:
maybe just a preferential thing, but I like to put `handle` at the top of management commands, and then you read downwards into the helper functions.